maint/gen_binding_c.py

```diff
@@ -5,28 +5,47 @@
 
 from local_python import MPI_API_Global as G
 from local_python.mpi_api import load_mpi_api
-from local_python.binding_c import dump_mpi_c, dump_Makefile_mk
+from local_python.binding_c import *
 from local_python import RE
-import os
 import glob
 
 def main():
+    # -- Loading APIs --
+    print("Loading maint/mpi_standard_api.txt ...")
     load_mpi_api("maint/mpi_standard_api.txt")
 
-    os.chdir("src/binding/c")
-    api_files = glob.glob("*_api.txt")
+    binding_dir = "src/binding/c"
+    api_files = glob.glob("%s/*_api.txt" % binding_dir)
     for f in api_files:
-        if RE.match(r'(\w+)_api.txt', f):
+        if RE.match(r'.*\/(\w+)_api.txt', f):
             # The name in eg pt2pt_api.txt indicates the output folder.
             # Only the api functions with output folder will get generated.
             # This allows simple control of what functions to generate.
+            print("Loading %s ..." % f)
             load_mpi_api(f, RE.m.group(1))
 
-    for func in G.FUNCS.values():
-        if 'dir' in func:
-            dump_mpi_c(func)
+    # -- Generating code --
+    func_list = [f for f in G.FUNCS.values() if 'dir' in f]
+    func_list.sort(key = lambda f: f['dir'])
+    for func in func_list:
+        if RE.search(r'not_implemented', func['attrs']):
```

Comment:
IMO, regex is overkill for simple substring matches. This can be done with `in` or `.find()`, with the benefit that it should perform better.

Reply:
I can change that.

Regex is fast enough, so I wouldn't worry about performance optimization unless there is evidence and justification. One of the drawback is regex tend to become too complex. That can be fixed by restructure the code design to keep the regex simple.

The benefit of regex vs direct string comparison or `in` or `find` is flexibility. For example, it can easily accommodate extra spaces, multiple spelling variations, flexible cases, etc. without significantly change the code appearance or structure. The alternative of using a mix of `==`, `in`, `find`, and regex, imo, can introduce unnecessary visual code complexity.

So I hope I convinced you to overlook this detail, but if you insist, I'll change it (and potentially many other places).